api/main_endpoints/routes/User.js

```diff
@@ -406,4 +408,104 @@ router.post('/getUserById', async (req, res) => {
   });
 });
 
+router.get('/isUserSubscribed', (req, res) => {
+  User.findOne({ email: req.query.email }, function(error, result) {
+    if (error) {
+      res.status(BAD_REQUEST).send({ message: 'Bad Request.' });
+    }
+
+    if (!result) {
+      return res
+        .status(NOT_FOUND)
+        .send({ message: `${req.query.email} not found.` });
+    }
+    if (result.emailOptIn)
+      return res.status(OK).send({ message: 'Subscribed!' });
+    else return res.status(OK).send({ message: 'No!' });
+  });
+});
+
+router.post('/setUserEmailPreference', (req, res) => {
+  const email = req.body.email;
+  const emailOptIn = req.body.emailOptIn;
```

Comment:
is this supposed to be a true/false value?

Reply:
the email is supposed to be an email and the email opt in is supposed to be a boolean